easybuild/framework/easyblock.py

```diff
@@ -2795,3 +2793,127 @@ class StopException(Exception):
     StopException class definition.
     """
     pass
+
+
+def inject_checksums(ecs, checksum_type):
+    """
+    Inject checksums of given type in specified easyconfig files
+
+    :param ecs: list of EasyConfig instances to inject checksums into corresponding files
+    :param checksum_type: type of checksum to use
+    """
+    for ec in ecs:
+        ec_fn = os.path.basename(ec['spec'])
+        print_msg("injecting %s checksums in %s" % (checksum_type, ec['spec']))
+
+        # get easyblock instance and make sure all sources/patches are available by running fetch_step
+        print_msg("fetching sources & patches for %s..." % ec_fn)
+        app = get_easyblock_instance(ec)
+        app.update_config_template_run_step()
+        app.fetch_step()
+
+        # check for any existing checksums, require --force to overwrite them
+        found_checksums = bool(app.cfg['checksums'])
+        for ext in app.exts:
+            found_checksums |= bool(ext.get('checksums'))
+        if found_checksums:
+            if build_option('force'):
+                print_warning("Found existing checksums in %s, overwriting them (due to use of --force)..." % ec_fn)
+            else:
+                raise EasyBuildError("Found existing checksums, use --force to overwrite them")
+
+        # back up easyconfig file before injecting checksums
+        ectxt = read_file(ec['spec'])
+        ec_backup = os.path.join(ec['spec'] + '.bck')
```

Comment:
changed to `.bak` after checking with @damianam, since he chose `.bck` initially via #2134